genie-web/src/main/java/com/netflix/genie/web/tasks/leader/ArchiveStatusCleanupTask.java

```diff
@@ -0,0 +1,140 @@
+import com.netflix.genie.web.data.services.PersistenceService;
+import com.netflix.genie.web.exceptions.checked.NotFoundException;
+import com.netflix.genie.web.properties.ArchiveStatusCleanupProperties;
+import com.netflix.genie.web.tasks.GenieTaskScheduleType;
+import com.netflix.genie.web.util.MetricsUtils;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import lombok.extern.slf4j.Slf4j;
+
+import java.time.Instant;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Leader task that find jobs whose archival status was left in 'PENDING' state.
+ * This can for example happen if the agent fails to update the server after successfully archiving.
+ * The logic is summarized as:
+ * If a job finished running more than N minutes ago, and the agent is disconnected and the archive status is PENDING,
+ * then set the archive status to UNKNOWN.
+ * This task ignores non-agent jobs because {@link ClusterCheckerTask} handles them already.
+ *
+ * @author mprimi
+ * @since 4.0.0
+ */
+@Slf4j
+public class ArchiveStatusCleanupTask extends LeaderTask {
+
+    private static final String CLEAR_ARCHIVE_STATUS_COUNTER_NAME = "genie.jobs.archiveStatus.cleanup.counter";
+    private static final String CLEAR_ARCHIVE_STATUS_TIMER_NAME = "genie.tasks.archiveStatusCleanup.timer";
+    private final PersistenceService persistenceService;
+    private final AgentRoutingService agentRoutingService;
+    private final ArchiveStatusCleanupProperties properties;
+    private final MeterRegistry registry;
+
+    /**
+     * Constructor.
+     *
+     * @param dataServices        data services
+     * @param agentRoutingService agent routing service
+     * @param properties          task properties
+     * @param registry            metrics registry
+     */
+    public ArchiveStatusCleanupTask(
+        final DataServices dataServices,
+        final AgentRoutingService agentRoutingService,
+        final ArchiveStatusCleanupProperties properties,
+        final MeterRegistry registry
+    ) {
+        this.persistenceService = dataServices.getPersistenceService();
+        this.agentRoutingService = agentRoutingService;
+        this.properties = properties;
+        this.registry = registry;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void run() {
+        final Set<Tag> tags = Sets.newHashSet();
+        final long start = System.nanoTime();
+        try {
+            final Instant updatedThreshold = Instant.now().minus(this.properties.getGracePeriod());
+            final Set<String> jobIds =
+                this.persistenceService.getFinishedJobsWithPendingArchiveStatus(updatedThreshold);
+            if (!jobIds.isEmpty()) {
+                log.debug("Found {} finished jobs with PENDING archive status", jobIds.size());
+                this.clearJobsArchiveStatus(jobIds);
+            }
+            MetricsUtils.addSuccessTags(tags);
```

Comment:
Not specific to this PR but I wonder if we can roll this adding of success or failure tags into an aspect somehow so we don't have to write the same thing all over the place.

Reply:
Would be nice, maybe annotate such metods with `@GenieMetric(name="...")` or something like that.